src/test/common/clara/test_memory.cljc

```diff
@@ -47,8 +49,41 @@
 ;; the intent of these tests is to create patterns in the engine that cover edge cases and other paths
 ;; of concern in clara.rules.memory.
 
-(def-rules-test test-query-for-many-added-elements
+(def-rules-test test-negation-complex-join-with-numerous-non-matching-facts-inserted-after-descendant-negation
+  {:rules []
+   :queries [query1 [[]
+                     [[Hot (= ?t temperature)]
+                      [:not [Cold (tu/join-filter-equals ?t temperature)]]]]]
+   :sessions [empty-session [query1] {}]}
+  (let [lots-of-hot (doall (for [_ (range 100)]
+                             (->Hot 20)))]
+    (is (= (repeat 100 {:?t 20})
+           (-> empty-session
+               (insert (->Cold 10))
+               fire-rules
+               (insert-all lots-of-hot)
+               fire-rules
+               (query query1))))))
+;; These tests are here to verify https://github.com/cerner/clara-rules/issues/303
```

Comment:
Can we merge these two new tests into a single test body as discussed at https://github.com/cerner/clara-rules/pull/334#discussion_r133224727? Mainly that will clarify what these comments apply to.

Reply:
Merged the two tests --> https://github.com/ris5438/clara-rules/commit/848f342f9544e630adb4d40f55af8dae7753e4d3